docs/gcloud-auth.rst

```diff
@@ -58,15 +58,66 @@ properties of your local environment in the following order
 #. Credentials provided by the Google Cloud SDK (via ``gcloud auth login``)
 #. Application running in Google Compute Engine
 
-Loading Credentials Explicitly
-------------------------------
+Explicit Credentials
+====================
 
-In addition, the
+The Application Default Credentials discussed above can be useful
+if your code needs to run in many different environments or
+if you just don't want authentication to be a focus in your code.
+
+However, if your code will only run in one place, you may want to
```

Comment:
Sorry - - my thought was along the lines of:

I have a script that does some analytical stuff on our data. I don't ever want to run that as "me", I always want to run as "the analytics user". I want everyone on my team to be able to run that, and count toward that users quota... You might want to hard-code the credentials into that internal script, so that everyone can run under the same bucket.

Not sure if that jives with what you're saying here, but that's a case where I might consider explicitly setting credentials in my code.